vita3k/gui/src/user_management.cpp

```diff
@@ -18,212 +18,483 @@
 #include "private.h"
 
 #include <config/functions.h>
-#include <host/state.h>
+#include <gui/functions.h>
 #include <misc/cpp/imgui_stdlib.h>
+#include <util/log.h>
+#include <util/string_utils.h>
+
+#include <nfd.h>
+#include <pugixml.hpp>
+#include <stb_image.h>
 
 namespace gui {
 
-void init_user(GuiState &gui, HostState &host, const std::string &user_id) {
-    host.io.user_id = user_id;
-    host.io.user_name = host.cfg.online_id[std::stoi(user_id)];
+static bool init_avatar(GuiState &gui, HostState &host, const std::string &user_id, const std::string &avatar) {
```

Comment:
This function shares lots of code with `init_user_background`, can you abstract it in some way?

Reply:
diferent header, i load user background only for one user, like theme, avatar is load for all user on boot, totaly diferent